app/services/users/validate_reset_token_service.rb

```diff
@@ -22,10 +22,31 @@ def user
     def valid_request?
       return false if user.blank?
 
-      return false if user.reset_password_sent_at.blank?
+      return false if user.reset_password_sent_at.blank? && time_limitation?
 
+      if time_limitation?
+        valid_time?
+      else
+        true
+      end
+    end
+
+    def valid_time?
       time_since_last_mail = Time.zone.now - user.reset_password_sent_at
-      time_since_last_mail < 30.minutes
+      time_since_last_mail < time_limit_minutes
+    end
+
+    def time_limitation?
+      ENV.fetch('RESET_PASSWORD_TOKEN_TIME_LIMIT') { '' }.present?
+    end
+
+    def time_limit_minutes
+      env_var = ENV.fetch('RESET_PASSWORD_TOKEN_TIME_LIMIT') { '' } 
+      if env_var.present?
+        env_var.to_i.minutes
+      else
+        0.minutes
```

Comment:
why if we have the `time_limitation?` check ? this should not be possible

Reply:
Removed when rebased from master.